Code/GraphMol/MolInterchange/Parser.cpp

```diff
@@ -970,10 +970,13 @@ std::vector<boost::shared_ptr<ROMol>> DocToMols(
     }
     for (const auto &molval : doc["molecules"].GetArray()) {
       auto *mol = new RWMol();
+
+      // Add the mol to res, so that it is managed in case processMol throws
```

Comment:
This changes behavior in the case that there is a problem
I'd prefer to handle the chance of a leak by making `mol` a `std::unique_ptr<RWMol>` and changing line 976 to:
`res.emplace_back(static_cast<ROMol *>(mol.release()))`

Reply:
Ok